xds/src/test/java/io/grpc/xds/GrpcXdsClientImplDataTest.java

```diff
@@ -2613,6 +2614,87 @@ public void validateCommonTlsContext_validationContextProviderInstance()
         .validateCommonTlsContext(commonTlsContext, ImmutableSet.of("name1", "name2"), false);
   }
 
+  @Test
+  public void
+      validateCommonTlsContext_combinedValidationContextSystemRootCerts_envVarNotSet_throws() {
+    XdsClusterResource.enableSystemRootCerts = false;
```

Comment:
These don't restore the value. You can either try-finally, or you can have a `@Before` that saves the value and an `@After` that sets the value to the saved value (independent of whether it changed).

Reply:
Saving and restoring the flag value in @Before and @After now.